### ISSUE TITLE
Resource status restore doesn't work for resources without status subresource

### ISSUE THREAD

**Issue by Lyndon-Li** [vmware-tanzu/velero#6016]
As Velero workflow, the status fields in the object to be restored will be removed before creating the object. This is done for all kinds of objects.
There is a feature to allow users selectively restore status. 
Therefore, after the objects a created, if the status restore is required, Velero calls UpdateStatus to update the status to the created object.

However, this `UpdateStatus` approach only works for the resources that enable `status` subresource, if the `status` subresource is not enabled, the call to `UpdateStatus` will fail.

Conclusively:
A object doesn't enable status subresource
Users select to restore status for the object
The status restore will finally fail

**Comment by Lyndon-Li** [comment]
> Hmm. So if there is no status subresource, would you be able to patch status using the regular endpoint for the resource?

Yes. So for status subresource, we need to use UpdateStatus; for non status subresource, we need to use regular patch.

**Comment by blackpiglet** [comment]
unstale
